### PR TITLE
DataDogService: set default start_time value in case it is nil

### PR DESCRIPTION
**Pull request by yoomlam**
Resolves `PushPriorityAppealsToJudgesJob` not being able to report error - see https://dsva.slack.com/archives/CJL810329/p1641301977217200?thread_ts=1641301718.217100&cid=CJL810329

### Description
In `DataDogService`, set a default value start_time in case it is nil.

### Acceptance Criteria
- [ ] Code compiles correctly

### Testing Plan
Deploy to prod and rerun job to see if we can get an error message revealing the job's problem

### CODE REVIEW

File: app/services/data_dog_service.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@ def self.increment_counter(metric_group:, metric_name:, app_name:, attrs: {}, by
     @statsd.increment(stat_name, tags: tags, by: by)
   end
 
-  def self.record_runtime(metric_group:, app_name:, start_time:)
+  def self.record_runtime(metric_group:, app_name:, start_time: Time.zone.now)
     metric_name = "runtime"
     job_duration_seconds = Time.zone.now - start_time
```
Comment (yoomlam) [inline app/services/data_dog_service.rb:16]: Ruby is trying to call the unary `-` on a nil `start_time`, which produces a `NoMethodError: undefined method `-@' for nil:NilClass`

File: spec/feature/queue/case_details_spec.rb [hunk 2]
```diff
@@ -1745,7 +1745,7 @@ def wait_for_page_render
 
         let(:veteran_full_name) { veteran.first_name + veteran.last_name }
         let(:nod_date) { "11/11/2020" }
-        let(:later_nod_date) { Time.zone.now.next_year(2).mdY }
+        let(:later_nod_date) { Time.zone.now.next_year(3).mdY }
```
Comment (yoomlam) [inline spec/feature/queue/case_details_spec.rb:1748]: Temporary fix until a better fix is merged: https://dsva.slack.com/archives/C2ZAMLK88/p1641322898345300?thread_ts=1641321913.342600&cid=C2ZAMLK88

Comment (yoomlam) [reply spec/feature/queue/case_details_spec.rb:1748]: https://github.com/department-of-veterans-affairs/caseflow/pull/17074

Comment (jcq) [reply spec/feature/queue/case_details_spec.rb:1748]: Is this a fix for unrelated flake?

**Review by jcq — APPROVED** [review]
LGTM — fixes the immediate problem of the `nil` for `start_time`

**Pull request merged**